servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TerminalSignalConsumerMock.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import org.mockito.Mockito;
+
+import static org.mockito.AdditionalAnswers.delegatesTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Mocked {@link TerminalSignalConsumer} that helps to verify that only a single terminal method was invoked.
+ */
+public final class TerminalSignalConsumerMock implements TerminalSignalConsumer {
```

Comment:
I am not really sure about this class if you remove the `Runnable` reference and use `Mockito.verifyNoMoreInteractions(signalConsumer);`. It is more intuitive if the verification is done inside the test as opposed to inside a common method like this.

Reply:
Agreed, if I will have a different `TerminalSignalConsumer` for `Single` it will have even less sense. Let me remove it.